_posts/2017-11-21-dotnet-core-rest-api-in-about-30-minutes.md

```diff
@@ -0,0 +1,587 @@
+  * `ToAsyncEnumerable()`: Modifies the DB call to tell it to perform the query asynchronously. Returns the `IAsyncEnumerable`, which is an EF Core construct that causes any method that materializes the results to be asynchronous.
+  * `Task<IEnumerable<Airplane>>`: Returns a Task allowing the calling method (the guts of ASP .NET Core, in this case) to await the results.
+
+***
+* POST
+
+  Now, before we can do much more, we need to be able to create a record. Add the following to your class:
+
+  ```c#
+  // POST api/airplanes
+  [HttpPost]
+  public async Task Post([FromBody]Airplane request)
+  {
+      _context.Add(request);
+      await _context.SaveChangesAsync();
+  }
+  ```
+
+  This method will take the `json` representing an Airplane entity from the body of the request, add it to the context, and then save the changes of the added entity to the context, thus adding it to the In-Memory database. Try it out, by opening Postman, and doing the following:
+  * Open a new request by clicking the + icon if you already have a tab open, or simply use the existing tab
+  * Use the dropdown where it says "GET" and change it to "POST".
+  * To the right of the dropdown, enter the URL: `http://localhost:5000/api/airplanes`
+  * Below that, click `Body`, select `raw`, and then click where it says `Text` and instead select `JSON (application/json)`
+  * In the text area for the request body, enter the following:
+  ```json
+  {
+      "ModelNumber": "AZ-999",
+      "SerialNumber": "12345",
+      "CurrentAirportCode": "BWI",
+      "SeatCount": 232,
+      "WeightInKilos": 1234567
+  }
+  ```
+  * Click "Send"
+
+  Above the bottom panel, to the far right, you should see: `Status: 200 OK`, along with the request time. If you do not see 200, there was an error. Check the console log to see if you can fix the error. If it was added correctly, you should be able to open up a new Postman tab, leave it as a `GET`, and enter the URL: `http://localhost:5000/api/airplanes` to see the results containing the Airplane you just added.
+
+  ***
+  :question:
+
+  You'll probably be thinking, "Didn't we specifically make the empty constructor and all the properties of Airplane private, so calling code would have to use the constructor? How did we create it without doing that?". Well yes, normally we wouldn't be asking for an actual EF entity in the request, or returning it back to the client. Usually we would use Request/Response "ViewModels" for the API endpoint layer. 
+  
+  However, for the sake of simplicity in the initial setup, we'll just use the entity. The .NET Core default JSON serializer allows private properties, so it will work for now. In future posts I will go over using a Service layer, containing "ViewModels", to prevent the client from being able to control the entity's state.
+  ***
+
+***
+* GET (one)
```

Comment:
having this in a list pushes the style to the right a bit in the output, would having `#### GET (one) allow for the breakup that is needed but not have the indent that causes the style to get off?

Reply:
That was actually intentional (to be a list under "More API Methods"), but as with some of the other things, it looked a lot better in VS Code Preview ☹️

Reply:
Yeah it is so different, the m ⬇️ preview it is spot on but when viewed in the site things get out of whack.